src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -739,6 +739,67 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   librados::IoCtx &io_ctx;
   librbd::Image &image;
   unsigned long size;
+  std::thread notify_thread;
+  ceph::condition_variable cond;
+  ceph::mutex lock = ceph::make_mutex("NBDWatchCtx::Locker");
+  bool notify = false;
+  bool terminated = false;
+
+  bool wait_notify() {
+    dout(10) << __func__ << dendl;
+
+    std::unique_lock locker{lock};
+    cond.wait(locker, [this] { return notify || terminated; });
+
+    notify = false;
```

Comment:
This would make more sense after "got notify request" dout IMO.  At this point you don't know whether you were notified or terminated -- not that it matters much, but still... ;)